### ISSUE TITLE
Gmail Unread Indicator Inconsistency

### ISSUE THREAD

**Issue by sspohn** [wavebox/waveboxapp#63]
* Which version of Wavebox are you using? [e.g. 3.0.0]
3.1.3
* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
macOS 10.12.4

I'm having some inconsistency with unread indicators on Gmail accounts. I had Wavebox Pro set up with three Gmail accounts (two gmail.com, one Google Apps domain) - all three of them were failing to show the unread indicator when new messages were received. I also had a Trello account and two Slack accounts set up - those unread indicators were working properly. Since I wasn't sure if this was an issue carried over from my previous 3.1.2 installation, I removed the ~/Library/Application Support/wavebox folder, and set up Wavebox again fresh on 3.1.3, this time importing my accounts from Wmail. Once I reauthenticated, the Google Apps account is showing unread messages correctly, but the Gmail accounts are not. I've made sure that the inbox type is set correctly as shown at https://github.com/wavebox/waveboxapp/wiki/KB:-Google-Unread-Counts. Anything I can do to help with logs, please let me know!

**Comment by tcm214** [comment]
I have the opposite issue this morning.  All day it has insisted I have an unread email in gmail.  I tried reloading it and no change. 
![wavebox bug](https://cloud.githubusercontent.com/assets/25810509/24968960/550b6458-1f6c-11e7-8551-a1fe38d88aa0.PNG)

**Comment by Thomas101** [comment]
@sspohn sorry to hear they've been a bit flakey today. If you right click on the account in the left and press `Resync` does this bring it back up to date? Do you mind me asking what the unread mode that you've used for these accounts?

@tcm214 have you tried the steps in the [KB article?](https://github.com/wavebox/waveboxapp/wiki/KB:-Google-Unread-Counts)

**Comment by sspohn** [comment]
Neither reload nor resync have an impact on the unread indicator. I thought I was using Unread Messages in Primary Category on all of my accounts - but it turns out that the Google Apps account was just set to Unread Messages. If I change it to Unread Messages in Primary Category, the unread indicator disappears. If I change it back to Unread Messages, it comes back. The same thing happens on my other accounts - so it seems like Unread Messages in Primary Category isn't working quite right. All three accounts are set to the Default inbox type in Gmail.

**Comment by tcm214** [comment]
![wavebox bug2](https://cloud.githubusercontent.com/assets/25810509/24972320/4ac623e6-1f78-11e7-8362-f8c8cd60f4b7.PNG)

I went in and changed my wavebox settings to the correct option for gmail 'default'.  Now I have the opposite problem haha

**Comment by Thomas101** [comment]
@sspohn if you paste the following into the gmail search bar do you see your unread messages? `label:inbox label:unread category:important`

@tcm214 by the screenshot it looks like you're using the `Important First` gmail inbox. Have you tried setting the unread mode to `Unread Important Messages`?

**Comment by sspohn** [comment]
Yes, I do see the unread messages. To be precise, I see them from all of the Gmail tabs (including Social, Promotions, and Updates). If I do `label:inbox label:unread category:primary` I see the unread messages on just the Primary tab.

**Comment by Thomas101** [comment]
Thanks, if you go to View > Developer Tools and then click on the console tab and paste the following 
```js
(function () {
const data = require('electron').ipcRenderer.sendSync(`storageBucket:mailboxes:allItems`, { sync: true }).response
const log = Object.keys(data).map((k) => {
  if (k === '__index__') { return JSON.parse(data[k]) }
  return {id: k, s: JSON.parse(data[k]).services.map((s) => s.type + ':' + s.unreadMode + ':' + s.accessMode) }
})
console.log(JSON.stringify(log))
})()
```
could you paste what comes out from that? It should be something similar to...

```
[["f1d983c2-e218-40c0-a352-00b1057ed5fc","ae6cb26a-3806-4951-aa92-23951b07473f","6fb489e9-eb10-4b7b-a976-4271a04ae271","3adc1b7a-f579-4e38-bc47-839433d51f21","03a257e8-c5c2-4cdd-bbae-fc260a34d4ba"],{"id":"ae6cb26a-3806-4951-aa92-23951b07473f","s":["DEFAULT:INBOX_UNREAD_UNBUNDLED:GINBOX"]},{"id":"f1d983c2-e218-40c0-a352-00b1057ed5fc","s":["DEFAULT:INBOX_UNREAD_UNBUNDLED:GINBOX","CALENDAR:undefined:undefined","STORAGE:undefined:undefined","CONTACTS:undefined:undefined","COMMUNICATION:undefined:undefined"]},{"id":"6fb489e9-eb10-4b7b-a976-4271a04ae271","s":["DEFAULT:INBOX_UNREAD_UNBUNDLED:GINBOX"]},{"id":"3adc1b7a-f579-4e38-bc47-839433d51f21","s":["DEFAULT:undefined:undefined"]},{"id":"03a257e8-c5c2-4cdd-bbae-fc260a34d4ba","s":["DEFAULT:undefined:undefined"]}]
```

Thanks

**Comment by sspohn** [comment]
Here you go - I redacted the UUIDs of my accounts:

`[{"id":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","s":["DEFAULT:INBOX_UNREAD_PERSONAL:GMAIL","CALENDAR:undefined:undefined","STORAGE:undefined:undefined","NOTES:undefined:undefined"]},["xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"],{"id":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","s":["DEFAULT:INBOX_UNREAD_PERSONAL:GMAIL","CALENDAR:undefined:undefined","STORAGE:undefined:undefined","NOTES:undefined:undefined"]},{"id":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","s":["DEFAULT:INBOX_UNREAD_PERSONAL:GMAIL","CALENDAR:undefined:undefined","STORAGE:undefined:undefined","NOTES:undefined:undefined"]}]`

**Comment by tcm214** [comment]
Hey Thomas,  I think I just have gmail on Default.  This morning it's still acting up.
![waveboxbug3](https://cloud.githubusercontent.com/assets/25810509/25011464/afed6a7e-202a-11e7-81aa-eba3277aca63.PNG)
![waveboxbug4](https://cloud.githubusercontent.com/assets/25810509/25011468/b336c8e2-202a-11e7-9f74-d9e6b4fbc00c.PNG)
![waveboxbug5](https://cloud.githubusercontent.com/assets/25810509/25011473/b66318ea-202a-11e7-9c62-6d100bb21094.PNG)

**Comment by Thomas101** [comment]
Looks like I've found it!

**Comment by Thomas101** [comment]
This has been fixed in 3.1.4 which is now live. Would you be able to give it another try? Thanks

Wavebox should update automatically after a while, but if you want to hurry it up, just restart the app and it will re-check

**Comment by sspohn** [comment]
I don't have my Mac handy, but it's working as intended on Fedora 25 w/ 3.1.4. Thanks for the fix!
